src/opm/output/eclipse/AggregateMSWData.cpp

```diff
@@ -489,6 +501,61 @@ namespace {
             };
         }
 
+        float valveFlowUnitCoefficient(const Opm::UnitSystem::UnitType uType)
+        {
+            using UType = Opm::UnitSystem::UnitType;
+
+            switch (uType) {
+                case UType::UNIT_TYPE_METRIC:
+                    return 1.340e-15f;
+
+                case UType::UNIT_TYPE_FIELD:
+                    return 2.892e-14f;
+
+                case UType::UNIT_TYPE_LAB:
+                    return 7.615e-14f;
+
+                case UType::UNIT_TYPE_PVT_M:
+                    return 1.322e-15f;
+            }
+
+            throw std::invalid_argument {
+                "Unsupported Unit Convention: '" +
+                std::to_string(static_cast<int>(uType)) + '\''
+            };
+        }
+
+        template <class RSegArray>
+        void assignValveCharacteristics(const ::Opm::Segment&    segment,
```

Comment:
This function contains calculations similar or equivalent to those in opm-simulators. Why are they necessary here? If we need to output and/or restart from these quantities, should be get them from the simulator code instead, to be consistent with what is actually used?

Reply:
Not really.  Getting them from the simulator means that the unit conversion becomes more difficult, at least for the `DeviceBaseStrength` which involves `Cu`.